# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/ogg/OggCodecState.cpp
# Commit: 77b8b81f6f0a
# Full Hash: 77b8b81f6f0a4efde76822c22e4ffd0e366e52bb
# Author: alwu <alwu@mozilla.com>
# Date: 2023-09-09 09:28:47
# Description:
#   Bug 1852170 - check if the duration is valid. r=media-playback-reviewers,azebrowski
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D187803
# ==============================================================================

diff -r 61380efc6f72 -r 77b8b81f6f0a dom/media/ogg/OggCodecState.cpp
--- a/dom/media/ogg/OggCodecState.cpp	Sat Sep 09 02:24:40 2023 +0000
+++ b/dom/media/ogg/OggCodecState.cpp	Sat Sep 09 03:36:42 2023 +0000
@@ -273,7 +273,7 @@
   NS_ASSERTION(endTimestamp.IsPositiveOrZero(), "timestamp invalid");
 
   TimeUnit duration = PacketDuration(packet.get());
-  if (!duration.IsPositiveOrZero()) {
+  if (!duration.IsValid() || !duration.IsPositiveOrZero()) {
     NS_WARNING(
         nsPrintfCString("duration invalid! (%s)", duration.ToString().get())
             .get());
